mitmproxy/ctx.py

```diff
@@ -0,0 +1,12 @@
+from __future__ import absolute_import, print_function, division
+
+from mitmproxy import ctxmanager
+
+_flow = None
+_log = None
+_master = None
+
+
+flow = ctxmanager.Facade(lambda: _flow)
+log = ctxmanager.Facade(lambda: _log)
+master = ctxmanager.Facade(lambda: _master)
```

Comment:
No, because this allows users to do "from mitmproxy.ctx import master", and not be surprised. What we can do is change this interface to make this clearer by making the context objects callables, i.e. "master = mitmproxy.ctx.master()", but I'm unsure if this is worth it.